compiler+runtime/src/cpp/jank/runtime/obj/tagged_literal.cpp

```diff
@@ -0,0 +1,109 @@
+#include <jank/runtime/obj/tagged_literal.hpp>
+#include <jank/runtime/obj/keyword.hpp>
+#include <jank/runtime/rtti.hpp>
+#include <jank/runtime/context.hpp>
+
+namespace jank::runtime::obj
+{
+  tagged_literal::tagged_literal(object_ptr const tag, object_ptr const form)
+    : tag{ tag }
+    , form{ form }
+  {
+  }
+
+  native_bool tagged_literal::equal(object const &o) const
+  {
+    if(o.type != object_type::tagged_literal)
+    {
+      return false;
+    }
+
+    auto const s(expect_object<tagged_literal>(&o));
+    return runtime::equal(tag, s->tag) && runtime::equal(form, s->form);
+  }
+
+  static void
+  to_string_impl(object_ptr const &tag, object_ptr const &form, util::string_builder &buff)
+  {
+    buff('#');
+    to_string(tag, buff);
+    buff(' ');
+    to_string(form, buff);
+  }
+
+  void tagged_literal::to_string(util::string_builder &buff) const
+  {
+    to_string_impl(tag, form, buff);
+  }
+
+  native_persistent_string tagged_literal::to_string() const
+  {
+    util::string_builder buff;
+    to_string_impl(tag, form, buff);
+    return buff.release();
+  }
+
+  native_persistent_string tagged_literal::to_code_string() const
+  {
+    return to_string();
+  }
+
+  native_hash tagged_literal::to_hash() const
+  {
+    if(hash)
+    {
+      return hash;
+    }
+
+    return hash = hash::combine(hash::visit(tag), hash::visit(form));
+  }
+
+  object_ptr tagged_literal::get(object_ptr const key, object_ptr const fallback) const
+  {
+    auto const tag_kw{ __rt_ctx->intern_keyword("tag").expect_ok() };
+    auto const form_kw{ __rt_ctx->intern_keyword("form").expect_ok() };
+
+    if(tag_kw == key)
+    {
+      return tag;
+    }
+
+    if(form_kw == key)
+    {
+      return form;
+    }
+
+    return fallback;
+  }
+
+  object_ptr tagged_literal::get(object_ptr const key) const
+  {
+    return get(key, obj::nil::nil_const());
+  }
+
+  object_ptr tagged_literal::get_entry(object_ptr key) const
```

Comment:
Please make `key` const here and in `contains`.

Reply:
I'm not sure how I keep messing this up 🤦

Reply:
Most C++ devs don't even think about that. I'm just a stickler for const-correctness and have trained my eye for it. You'll get there!